testing/trino-tests/src/test/java/io/trino/execution/TestViewWithTableFunction.java

```diff
@@ -0,0 +1,91 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import io.trino.Session;
+import io.trino.plugin.base.security.AllowAllSystemAccessControl;
+import io.trino.plugin.jdbc.JdbcPlugin;
+import io.trino.plugin.jdbc.TestingH2JdbcModule;
+import io.trino.plugin.memory.MemoryPlugin;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SystemAccessControl;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingAccessControlManager;
+import io.trino.testing.TestingAccessControlManager.TestingPrivilege;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static io.trino.testing.TestingAccessControlManager.TestingPrivilegeType.GRANT_EXECUTE_FUNCTION;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestViewWithTableFunction
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        // TODO (https://github.com/trinodb/trino/issues/12833) switch to FileBasedSystemAccessControl once it supports access rules for table functions
+        SystemAccessControl accessControl = AllowAllSystemAccessControl.INSTANCE;
+        QueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION)
+                // Access control happens on the coordinator only, so we don't need multiple nodes
+                .setNodeCount(1)
+                .setSystemAccessControl(accessControl)
+                .build();
+        try {
+            queryRunner.installPlugin(new JdbcPlugin("base-jdbc", new TestingH2JdbcModule()));
+            queryRunner.createCatalog("jdbc", "base-jdbc", TestingH2JdbcModule.createProperties());
+
+            queryRunner.installPlugin(new MemoryPlugin());
+            queryRunner.createCatalog("memory", "memory", Map.of());
+
+            return queryRunner;
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+    }
+
+    @Test
+    public void testViewWithTableFunction()
+    {
+        Session viewOwner = getSession();
+        Session otherSession = Session.builder(getSession())
+                .setIdentity(Identity.ofUser(getSession().getUser() + "-someone-else"))
+                .build();
+
+        assertUpdate(viewOwner, "CREATE VIEW memory.default.view_with_ptf AS SELECT * FROM TABLE (jdbc.system.query('SELECT ''from h2'', monthname(CAST(''2005-09-10'' AS date))'))");
```

Comment:
Shouldn't it be two tests with `SECURITY DEFINER` and `SECURITY INVOKER` to cover both cases?